src/cr/cube/cube.py

```diff
@@ -547,19 +547,30 @@ def _slice_idxs(self):
             return (0,)
         return range(len(self.dimensions[0].valid_elements))
 
+    @lazyproperty
+    def _show_missing(self):
+        """True if cube dimensions doesn't contain MR and NUM_ARRAY else False."""
+        return not all(d in self.dimension_types for d in [DT.MR, DT.NUM_ARRAY])
+
     @lazyproperty
     def _valid_idxs(self):
         valid_idxs = np.ix_(
             *tuple(d.valid_elements.element_idxs for d in self._all_dimensions)
         )
-        # ---NOTE FOR FUTURE: We'll need a way to tell to the cube which dimensions we
-        # ---want transposed and which not. So the ::-1 can be converted in something
-        # ---more robust.
-        return (
-            valid_idxs[::-1]
-            if self._measure(self.is_weighted).requires_array_transposition
-            else valid_idxs
-        )
+
+        def _reshape_idxs(valid_indices):
+            if self._measure(self.is_weighted).requires_array_transposition:
+                if len(self._all_dimensions) == 3:
+                    # ---In case of 3D array and a numeric array is involved we have to
+                    # ---change the order of the valid idxs, from [0,1,2] to [1,2,0]
+                    return valid_indices[1], valid_indices[-1], valid_indices[0]
```

Comment:
I'm not sure that this is always true. We _could_ specify a different transform, in which case you would have to check it (the transform) in order to generate the right shape. We should also put a note here that we're expecting this in the future.

Reply:
i think that these 2 conditions matches that inversion operation:
```python
    if self._measure(self.is_weighted).requires_array_transposition:
        if len(self._all_dimensions) == 3:
```
I do agree that ATM this is an hack waiting for the dim_order ... but i'm not sure i follow your concern